### PR TITLE
adds e2e test for updates

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/1918.

### CODE REVIEW

File: service/controller/v15/cluster_resource_set.go [hunk 1]
```diff
@@ -364,53 +367,57 @@ func NewClusterResourceSet(config ClusterResourceSetConfig) (*controller.Resourc
...
+		}
+
+		statusResource, err = statusresource.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	resources := []controller.Resource{
+		statusResource,
```
Comment (rossf7) [inline service/controller/v15/cluster_resource_set.go]: <3

File: Gopkg.toml [hunk 2]
```diff
@@ -66,7 +66,7 @@ required = [
   name = "github.com/giantswarm/e2etemplates"
 
 [[constraint]]
-  branch = "master"
+  branch = "max"
   name = "github.com/giantswarm/e2etests"
```
Comment (xh3b4sd) [inline Gopkg.toml]: See https://github.com/giantswarm/e2etests/pull/11.

File: integration/test/update/main_test.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+			panic(err.Error())
+		}
+	}
+
+	{
+		c := update.Config{
+			Logger:   l,
+			Provider: p,
+
+			MaxWait: 90 * time.Minute,
```
Comment (kopiczko) [inline integration/test/update/main_test.go]: This is a bit terrifying.

**Comment by xh3b4sd** [comment]
> There is a problem with the e2e tests for updates on AWS. We assume that on each update all nodes have the new version. This implies rolling the nodes so far. We always roll them on Azure and KVM. We do not on AWS because CF is "smart".

https://gigantic.slack.com/archives/C7865GLSY/p1533828184000495

**Comment by rossf7** [comment]
Just to add my 2 cents. The ASG will roll the nodes if the launch config changes. There needs to be a launch config change that requires replacement.

For example having a generated ID in the cloudconfig somewhere. This is basically the same thing we do for the master node by setting the instance name.

**Comment by xh3b4sd** [comment]
We do not have launchconfigs for masters. We roll them "manually" because of the generated names in the CF templates. Just talked with Timo. Actually the user data should always change because we track the node versions via the labels we give to the kubelets so the nodes should always be rolled. I am checking now why this does not happen. Maybe some weird wiring issue.

File: service/controller/v15/cloudconfig/master_template.go [hunk 4]
```diff
@@ -55,6 +56,10 @@ func (c *CloudConfig) NewMasterTemplate(ctx context.Context, customObject v1alph
 		cloudConfigConfig.Params = params
 		cloudConfigConfig.Template = k8scloudconfig.MasterTemplate
 
+		fmt.Printf("\n")
+		fmt.Printf("%#v\n", cloudConfigConfig.Params.Cluster.Kubernetes.Kubelet.Labels)
+		fmt.Printf("\n")
```
Comment (TheoBrigitte) [inline service/controller/v15/cloudconfig/master_template.go]: Debugging ?

File: service/controller/v15/cloudconfig/worker_template.go [hunk 5]
```diff
@@ -45,6 +46,10 @@ func (c *CloudConfig) NewWorkerTemplate(ctx context.Context, customObject v1alph
 		cloudConfigConfig.Params = params
 		cloudConfigConfig.Template = k8scloudconfig.WorkerTemplate
 
+		fmt.Printf("\n")
+		fmt.Printf("%#v\n", cloudConfigConfig.Params.Cluster.Kubernetes.Kubelet.Labels)
+		fmt.Printf("\n")
```
Comment (TheoBrigitte) [inline service/controller/v15/cloudconfig/worker_template.go]: same here

**Comment by rossf7** [comment]
Yes I know we don't have launchconfigs for masters :)

Good point that the userdata should change due to the labels. So yes sounds like a wiring issue.

**Comment by xh3b4sd** [comment]
> We have an issue with the small CC. CF just interprets the user data it holds in its templates. Now this user data is only the small CC and when the version of the `k8scloudconfig` templates does not change nothing happens and the new version of the actual CC on S3 is not applied at all. It is just natural that the `k8scloudconfig` version does not necessarily change with each of our releases, because we may just do code changes in our operators. Should we hang again? Or should we for now just put some timestamp into the small CC to make it roll all the time so that we then optimize that later?

Just pasting from the Slack discussion. The idea now is to use the version of the operator's version bundle and not the one of the `k8scloudconfig` template for the bucket path. Any objections? I am going to explore that and probably create a PR based on this one so we can merge in series and have nicer diffs.

**Comment by kopiczko** [comment]
Makes sense to me.

**Comment by rossf7** [comment]
Yes I think changing the bucker path is a good approach.

File: integration/test/update/main_test.go (hunk 3 above)
Comment (xh3b4sd) [reply integration/test/update/main_test.go]: It takes up to 50 minutes usually as it seems. Having a buffer doesn't hurt.

**Comment by xh3b4sd** [comment]
CI was already green. Going.

**Pull request merged**